src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -4021,6 +4006,19 @@ public override BoundNode VisitAsOperator(BoundAsOperator node)
             return result;
         }
 
+        public override BoundNode VisitSuppressNullableWarningExpression(BoundSuppressNullableWarningExpression node)
+        {
+            var result = base.VisitSuppressNullableWarningExpression(node);
+
+            Debug.Assert(!IsConditionalState);
+            if (_performStaticNullChecks && this.State.Reachable)
+            {
```

Comment:
If the state when a variable is either uninitialized or in unreachable code were to be oblivious, I think that gives you the right semantics without having to check reachable state anywhere.

Reply:
Add `// PROTOTYPE` comment to address later.